### PR TITLE
makerepropkg download used buid config

### PR DESCRIPTION
**Pull request by jelly**

### CODE REVIEW

**Comment by jelly** [comment]
@anthraxx Be aware I force pushed your branch so the tests run.

**Comment by jelly** [comment]
## For testing this PR:

* Test building a new package with extra-x86_64-build records the build tool

```
cd /tmp/
archco filesystem
PATH=/home/jelle/projects/devtools/:$PATH extra-x86_64-build
tar xvf filesystem-2021.05.31-1-x86_64.pkg.tar.zst .BUILDINFO
```

Verify the buildtool was recorded:
```
[jelle@t14s][~/projects/devtools]%grep buildtool /tmp/filesystem/trunk/.BUILDINFO
buildtool = devtools
buildtoolver = 20210202-git
```

Now we want to reproduce this package, it should fail as the git version is unreleased:

```
[jelle@t14s][/tmp/filesystem/trunk]%/home/jelle/projects/devtools/makerepropkg filesystem-2021.05.31-1-x86_64.pkg.tar.zst
  -> retrieving 'devtools-20210202-git.pkg.tar.zst'...
curl: (22) The requested URL returned error: 404

  -> retrieving 'devtools-20210202-git.pkg.tar.xz'...
curl: (22) The requested URL returned error: 404

  -> retrieving 'devtools-20210202-git.pkg.tar'...
curl: (22) The requested URL returned error: 404

==> ERROR: failed to retrieve devtools-20210202-git
```


## Issues
What about existing packages with a buildtool defined?!! These will all break:
* We can't edit the BUILDINFO, breaks the signature
* We could fallback to the an older devtools version (hardcode the old release) so it fetches the correct .BUILDINFO
* We don't check the BUILDINFO format in this patch (I think it was updated to 2 in the last pacman)

File: makerepropkg.in [hunk 1]
```diff
@@ -185,7 +196,25 @@ for fname in "${installed[@]}"; do
         exit 1
     fi
 done
-printf '%s\n' "${allpkgfiles[@]}" | mkarchroot -M @pkgdatadir@/makepkg-x86_64.conf -U "${archroot_args[@]}" "${buildroot}"/root - || exit 1
+
+trap 'rm -rf $TEMPDIR' EXIT INT TERM QUIT
+TEMPDIR=$(mktemp -d --tmpdir makerepropkg.XXXXXXXXXX)
+
+makepkg_conf="${TEMPDIR}/makepkg.conf"
+# anything before buildtool support is pinned to last none buildtool devtools
```
Comment (jelly) [inline makerepropkg.in:213]: s/none/known/

File: makerepropkg.in [hunk 2]
```diff
@@ -172,23 +190,43 @@ BUILDTOOLVER="${buildinfo[buildtoolver]}"
...
+    get_makepkg_conf "devtools-20210202-3-any" "${makepkg_conf}"
+# prefere to assume devtools up until matching makepkg version so repository packages remain reproducible
+elif [[ "${BUILDTOOL}" = makepkg ]] && (( $(vercmp "${BUILDTOOLVER}" 6.0.1) <= 0 )); then
+    get_makepkg_conf "devtools-20210202-3-any" "${makepkg_conf}"
+# all devtools builds
+elif [[ "${BUILDTOOL}" = devtools ]]; then
+    get_makepkg_conf "${BUILDTOOL}-${BUILDTOOLVER}" "${makepkg_conf}"
+# fallback to current makepkg.conf
+else
+    makepkg_conf=@pkgdatadir@/makepkg-x86_64.conf
```
Comment (jelly) [inline makerepropkg.in:224]: This probably warrants a `msg2 falling back to default makepkg-x86_64.conf?

Or maybe to reflect my issue, what if I set an invalid buildtool, I'll never notice that it's wrong, so should it be printed explicitly?

**Review by jelly — COMMENTED** [review]
As I opened the PR I can't request changes or approve it, I think the PR just needs two small improvements and the new makerepropkg option should be documented in the man page.

Tested the changes locally and the fallback seems to work with a build package, setting a faulty version `BUILDTOOLVER=20210202-4-any` shows that the devtools package is trying to get fetched. So to me it looks good to go :)

File: makerepropkg.in (hunk 1 above)
Comment (anthraxx) [reply makerepropkg.in:213]: well i literally mean "none buildtool aware devtools" so gonna use "aware" here :)

**Comment by anthraxx** [comment]
merged via cli

**Pull request closed without merging**